example-app/App.tsx

```diff
@@ -40,6 +40,41 @@ const Header = () => {
 };
 
 const App = () => {
+  const [result, setResult] = React.useState<{
+    result: boolean | null;
+    errorReason: string | null;
```

Comment:
use `undefined` instead of `null`. It becomes complicated to read and maintain the code if we mix `undefined` and `null` everywhere.